scripts/population/load-regions

```diff
@@ -33,25 +56,71 @@ if [[ ! -e ${DATA_PATH}/$COUNTRY_CODE ]]; then
   exit 1
 fi
 
+# Simplification parameter
+# From mapshaper wiki https://github.com/mbloch/mapshaper/wiki/Command-Reference#-simplify
+#
+# interval= Specify simplification amount in units of distance.
+#           Uses meters when simplifying unprojected datasets in 3D space,
+#           otherwise uses the same units as the source data.
+#           In our case since our GeoJSON are not projected (ie. use WGS84 coordinates)
+#           then this means the resolution for the simplified shape will be in meters.
+
+INTERVAL_DISTANCE=${INTERVAL_DISTANCE-100}
+
 IFS=',';
 for i in $LEVELS; do
   FILE=${DATA_PATH}/${COUNTRY_CODE}/${COUNTRY_CODE}_adm${i}.geojson
   echo -n "   -> Processing $FILE ... "
-  psql -q -d $POSTGRES_DB -U $POSTGRES_USER -h $POSTGRES_HOST -p $POSTGRES_PORT << SQL_SCRIPT
-
-    WITH data AS (SELECT \$$`cat $FILE`\$$::json AS fc)
-    INSERT INTO "regions" (country, name, admin_level, the_geom)
-    SELECT
-      '${COUNTRY_NAME}',
-      feat#>>'{properties,name}' AS name,
-      ${i},
-      ST_SetSRID(ST_CollectionExtract(ST_Multi(ST_GeomFromGeoJSON(feat->>'geometry')), 3), 4326) as the_geom
-    FROM (
-      SELECT json_array_elements(fc->'features') AS feat
-      FROM data
-    ) AS f;
+
+  OFILE=$FILE
+  if [[ ! -z $INTERVAL_DISTANCE ]]; then
+    # Simplify using distance
+    OFILE=${DATA_PATH}/${COUNTRY_CODE}/${COUNTRY_CODE}_adm${i}_${INTERVAL_DISTANCE}.geojson
+
+    mapshaper $FILE -simplify interval=$INTERVAL_DISTANCE -o $OFILE
+
+    SIZE=$(stat --printf="%s" $FILE)
+    OSIZE=$(stat --printf="%s" $OFILE)
+
+    echo "   -> Finished file size went from $SIZE to $OSIZE"
+  fi
+
+  if [ "$FORCE" = "1" ]; then
+    psql -q -d $POSTGRES_DB -U $POSTGRES_USER -h $POSTGRES_HOST -p $POSTGRES_PORT << SQL_SCRIPT
+
+      WITH data AS (SELECT \$$`cat $OFILE`\$$::json AS fc)
+      UPDATE "regions" SET the_geom=f.the_geom
+      FROM (
+        SELECT
+          '${COUNTRY_NAME}' as country,
+          feat#>>'{properties,name}' AS name,
+          ${i} as admin_level,
+          ST_SetSRID(ST_CollectionExtract(ST_Multi(ST_GeomFromGeoJSON(feat->>'geometry')), 3), 4326) as the_geom
+        FROM (
+          SELECT json_array_elements(fc->'features') AS feat
+          FROM data
+        ) AS dummy
+      ) AS f
+      WHERE regions.country=f.country AND regions.name=f.name AND regions.admin_level=f.admin_level;
+
+SQL_SCRIPT
+  else
+    psql -q -d $POSTGRES_DB -U $POSTGRES_USER -h $POSTGRES_HOST -p $POSTGRES_PORT << SQL_SCRIPT
+
+      WITH data AS (SELECT \$$`cat $OFILE`\$$::json AS fc)
+      INSERT INTO "regions" (country, name, admin_level, the_geom)
+      SELECT
+        '${COUNTRY_NAME}',
+        feat#>>'{properties,name}' AS name,
+        ${i},
+        ST_SetSRID(ST_CollectionExtract(ST_Multi(ST_GeomFromGeoJSON(feat->>'geometry')), 3), 4326) as the_geom
+      FROM (
+        SELECT json_array_elements(fc->'features') AS feat
+        FROM data
+      ) AS f;
```

Comment:
Could this be turned into a single "UPSERT" (ie, [`INSERT ... ON CONFLICT UPDATE`](https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT))? Forcing usually means we want to reach the end state regardless of the actual state - ie, don't care if the data exists or not, we want it to exist afterwards. So forcing on a non existent region should also succeed.

Reply:
I understand that the upset will require an index on (country,name,admin_level). Is that OK?